packages/cache-redis/src/services/redis-cache.ts

```diff
@@ -0,0 +1,83 @@
+import { Redis } from "ioredis"
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { RedisCacheModuleOptions } from "../types"
+
+const DEFAULT_CACHE_TIME = 30 // 30 seconds
+const EXPIRY_MODE = "EX" // "EX" stands for an expiry time in second
+
+type InjectedDependencies = {
+  redisClient: Redis
+}
+
+class RedisCacheService implements ICacheService {
+  protected TTL: number
+  protected readonly redis: Redis
+
+  constructor(
+    { redisClient }: InjectedDependencies,
+    options: RedisCacheModuleOptions = {},
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    this.redis = redisClient
+    this.TTL = options.defaultTTL || DEFAULT_CACHE_TIME
```

Comment:
thought(non-blocking): Should we expose a `namespace` option to group your Medusa cache entries? We could default to `medusa::` or something similar.

Reply:
I've added the namespace option with the default `medusa:` namespace as you suggested but I think a default namespace could be potentially easy to miss (e.g. user searches for `my-key*` instead of `medusa:my-key*`) so maybe just default to an empty string, wdyt?

Reply:
That's true. But if we're explicit about it in the README, we should be fine no?

Reply:
Sounds good, I added it to the Readme